src/proto/message_translation/power_frame_msg.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#ifdef PLATFORMIO_BUILD
+#include <proto/power_frame_msg.nanopb.h>
+#else
+#include "proto/power_frame_msg.pb.h"
+#include "proto/primitive/primitive_types.h"
+
+extern "C"
+{
+#include "proto/power_frame_msg.nanopb.h"
+}
+#endif
+
+// The nanopb generated size isn't c++ compatible so we redefine it here
+#undef TbotsProto_PowerFrame_size
+#define TbotsProto_PowerFrame_size                                                       \
+    std::max(TbotsProto_PowerControl_size, TbotsProto_PowerStatus_size) +                \
+        2 * sizeof(uint32_t) + sizeof(uint16_t)
```

Comment:
https://github.com/nanopb/nanopb_cpp

There is a cpp version of this, I am not sure if its a drop-in replacement

Redifining this macro is fine if the cpp nanopb is drastically different from what is currently implemented. 

It also seems like a fairly new project, so probably not worth it if its any more work then just changing the github URL and nanopb.h -> nanopb_cpp.h

Reply:
I believe it is also fixed in nanopb 4.x so ill try that

Reply:
migrating to 4.5.0 requires additional changes. filed #2592